### ISSUE TITLE
UnicodeEncodeError when using signals (even PII)

### ISSUE THREAD

**Issue by Alx-AI** [lilacai/lilac#608]
While running through the Quick Start guide, when running the PII detection signal, I get encode errors even while using the suggested OpenOrca dataset (on Windows 11 Python 3.10.4 FWIW):

P.S. Thank you for open sourcing such an incredible tool! 

```
Route error: http://localhost:5432/api/v1/concepts/
not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_utils.py", line 24, in custom_route_handler
    return await original_route_handler(request)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\fastapi\routing.py", line 241, in app
    raw_response = await run_endpoint_function(
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\fastapi\routing.py", line 169, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\starlette\concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_concept.py", line 32, in get_concepts
    return DISK_CONCEPT_DB.list(user)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\concepts\db_concept.py", line 379, in list
    namespace, name = root.split('/')[-2:]
ValueError: not enough values to unpack (expected 2, got 1)

Route error: http://localhost:5432/api/v1/concepts/
not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_utils.py", line 24, in custom_route_handler
    return await original_route_handler(request)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\fastapi\routing.py", line 241, in app
    raw_response = await run_endpoint_function(
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\fastapi\routing.py", line 169, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\starlette\concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_concept.py", line 32, in get_concepts
    return DISK_CONCEPT_DB.list(user)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\concepts\db_concept.py", line 379, in list
    namespace, name = root.split('/')[-2:]
ValueError: not enough values to unpack (expected 2, got 1)

Scheduling task "596c0b0836a34f3f92867d29a4c20d21": "[local/open-orca-100k] Compute signal "pii" on "question"".
[local/open-orca-100k] Compute signal "pii" on "question":   0%|                 | 34/100000 [00:01<1:00:33, 27.51it/s]
Computing signal "pii" on local/open-orca-100k:('question',) took 1.674s.
2023-08-28 10:51:36,215 - distributed.worker - WARNING - Compute Failed
Key:       596c0b0836a34f3f92867d29a4c20d21
Function:  _execute_task
args:      ('local', 'open-orca-100k', {'signal': {}, 'leaf_path': ('question',)}, '596c0b0836a34f3f92867d29a4c20d21')
kwargs:    {}
Exception: "UnicodeEncodeError('charmap', 'Bosna Hersek (BH) Başbakanı Adnan Terziç, Avrupa haritasında kara bir delik kalması halinde, AB ile bütünleşme sürecinin küresel anlamda tarihi bir örnek teşkil edemeyeceğini söyledi.\\r\\n\\r\\nTranslate to English\\r\\n\\r\\nEnglish:', 20, 21, 'character maps to <undefined>')"

2023-08-28 10:51:36,245 - distributed.client - ERROR - Error in callback <function TaskManager.execute.<locals>.<lambda> at 0x000001E98A79D2D0> of <Future: error, key: 596c0b0836a34f3f92867d29a4c20d21>:
Traceback (most recent call last):
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\distributed\client.py", line 411, in execute_callback
    fn(fut)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\tasks.py", line 199, in <lambda>
    lambda task_future: self._set_task_completed(task_id, task_future))
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\tasks.py", line 176, in _set_task_completed
    raise e
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\distributed\worker.py", line 2996, in apply_function_simple
    result = function(*args, **kwargs)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\tasks.py", line 217, in _execute_task
    task(*args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_dataset.py", line 103, in _task_compute_signal
    dataset.compute_signal(options.signal, options.leaf_path, task_step_id=(task_id, 0))
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\data\dataset_duckdb.py", line 369, in compute_signal
    select_rows_result = self.select_rows([ROWID, signal_col],
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\data\dataset_duckdb.py", line 1058, in select_rows
    signal_out_list = list(signal_out)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\tasks.py", line 258, in progress
    for t in tq:
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\tqdm\std.py", line 1182, in __iter__
    for obj in iterable:
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\data\dataset_utils.py", line 301, in sparse_to_dense_compute
    out = next(dense_output)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\signals\pii.py", line 50, in compute
    secrets = list(find_secrets(text))
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\signals\pii_secrets.py", line 142, in find_secrets
    fp.write(content)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\tempfile.py", line 483, in func_wrapper
    return func(*args, **kwargs)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u015f' in position 20: character maps to <undefined>
```

**Comment by Alx-AI** [comment]
Might be related; trying with another dataset and even searching gives me an error:

```Task completed "b486f31150434bfdae0d2294a5b9f6a2": "[local/arxiv_papers] Load dataset" in 2m8s.
Route error: http://localhost:5432/api/v1/concepts/
not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_utils.py", line 24, in custom_route_handler
    return await original_route_handler(request)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\fastapi\routing.py", line 241, in app
    raw_response = await run_endpoint_function(
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\fastapi\routing.py", line 169, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\starlette\concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_concept.py", line 32, in get_concepts
    return DISK_CONCEPT_DB.list(user)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\concepts\db_concept.py", line 379, in list
    namespace, name = root.split('/')[-2:]
ValueError: not enough values to unpack (expected 2, got 1)```

**Comment by nsthorat** [comment]
Thanks for filing, looks like this is an issue with windows path versus linux paths (we explicitly split the path by '/' which is backwards on windows).

I'll try to look into this ASAP

**Comment by dsmilkov** [comment]
I'll look into this now

**Comment by Alx-AI** [comment]
Pulled most recent version (which solved the issue of 0.0.0.0) but it did not solve this one. 

Additionally I'm getting the following ValueError on launch
```INFO:     Started server process [24004]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:5432 (Press CTRL+C to quit)
Route error: http://127.0.0.1:5432/api/v1/concepts/
not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_utils.py", line 24, in custom_route_handler
    return await original_route_handler(request)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\fastapi\routing.py", line 241, in app
    raw_response = await run_endpoint_function(
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\fastapi\routing.py", line 169, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\starlette\concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_concept.py", line 32, in get_concepts
    return DISK_CONCEPT_DB.list(user)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\concepts\db_concept.py", line 379, in list
    namespace, name = root.split('/')[-2:]
ValueError: not enough values to unpack (expected 2, got 1)```

**Comment by nsthorat** [comment]
We haven't released the new code yet, fixing our publish script then will update you when it's in the latest version :)

**Comment by Alx-AI** [comment]
It fixed most of the issues! 
I can now search and do more than I was able to before however, the unicode encode error persists when running the PII signal:

```
Computing signal "pii" on local/arxiv_papers:('text',) took 1.147s.
2023-08-29 15:12:13,637 - distributed.worker - WARNING - Compute Failed
Key:       a743802a8a7942aeae2bd14081959306
Function:  _execute_task
args:      ('local', 'arxiv_papers', {'signal': {}, 'leaf_path': ('text',)}, 'a743802a8a7942aeae2bd14081959306')
kwargs:    {}
Exception: "UnicodeEncodeError('charmap', 'Olof Heden and Denis S. Krotov∗', 30, 31, 'character maps to <undefined>')"

2023-08-29 15:12:13,691 - distributed.client - ERROR - Error in callback <function TaskManager.execute.<locals>.<lambda> at 0x00000222FA487520> of <Future: error, key: a743802a8a7942aeae2bd14081959306>:
Traceback (most recent call last):
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\distributed\client.py", line 411, in execute_callback
    fn(fut)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\tasks.py", line 199, in <lambda>
    lambda task_future: self._set_task_completed(task_id, task_future))
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\tasks.py", line 176, in _set_task_completed
    raise e
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\distributed\worker.py", line 2996, in apply_function_simple
    result = function(*args, **kwargs)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\tasks.py", line 217, in _execute_task
    task(*args)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\router_dataset.py", line 103, in _task_compute_signal
    dataset.compute_signal(options.signal, options.leaf_path, task_step_id=(task_id, 0))
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\data\dataset_duckdb.py", line 369, in compute_signal
    select_rows_result = self.select_rows([ROWID, signal_col],
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\data\dataset_duckdb.py", line 1060, in select_rows
    signal_out_list = list(signal_out)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\tasks.py", line 258, in progress
    for t in tq:
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\tqdm\std.py", line 1182, in __iter__
    for obj in iterable:
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\data\dataset_utils.py", line 301, in sparse_to_dense_compute
    out = next(dense_output)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\signals\pii.py", line 50, in compute
    secrets = list(find_secrets(text))
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\site-packages\lilac\signals\pii_secrets.py", line 142, in find_secrets
    fp.write(content)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\tempfile.py", line 483, in func_wrapper
    return func(*args, **kwargs)
  File "C:\Users\ajduf\anaconda3\envs\lilac\lib\encodings\cp1252.py", line 19, in encode
    return codecs.charmap_encode(input,self.errors,encoding_table)[0]
UnicodeEncodeError: 'charmap' codec can't encode character '\u2217' in position 30: character maps to <undefined>
```

**Comment by dsmilkov** [comment]
Thanks for the update -- I'm on it!

**Comment by dsmilkov** [comment]
Fixed with the 0.0.17 release
